tests/data/test_meta_dataset.py

```diff
@@ -0,0 +1,57 @@
+import json
+
+import pytest
+
+from mlflow.data.delta_dataset_source import DeltaDatasetSource
+from mlflow.data.http_dataset_source import HTTPDatasetSource
+from mlflow.data.huggingface_dataset_source import HuggingFaceDatasetSource
+from mlflow.data.meta_dataset import MetaDataset
+from mlflow.types import DataType
+from mlflow.types.schema import ColSpec, Schema
+
+
+@pytest.mark.parametrize(
+    ("dataset_source_class", "path"),
+    [
+        (HTTPDatasetSource, "test:/my/test/uri"),
+        (DeltaDatasetSource, "fake/path/to/delta"),
+        (HuggingFaceDatasetSource, "databricks/databricks-dolly-15k"),
+    ],
+)
+def test_create_meta_dataset_from_source(dataset_source_class, path):
+    source = dataset_source_class(path)
+    dataset = MetaDataset(source=source)
+
+    json_str = dataset.to_json()
+    parsed_json = json.loads(json_str)
+
+    assert parsed_json["digest"] is not None
```

Comment:
can we add some tests on the digest content itself? its important that different dataset sources will map to different digests

Reply:
makes sense to me, adding.